common/headtracker/types/head_saver.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import (
+	"context"
+
+	commontypes "github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// HeadSaver is an chain agnostic interface for saving and loading heads
+// Each chain will have its own custom implementation of HeadSaver interface.
+type HeadSaver[H commontypes.Head[BLOCK_HASH], BLOCK_HASH commontypes.Hashable] interface {
```

Comment:
Could this type be defined in `common/types` instead?
Doing this generally when possible would cut down on a lot of import aliases and number of imports in general.